spec/lib/chat_transcript_service_spec.rb

```diff
@@ -74,4 +74,46 @@ def service(message_ids)
     [/chat]
     MARKDOWN
   end
+
+  it "generates image / attachment / video / audio markdown inside the [chat] bbcode for upload-only messages" do
+    SiteSetting.authorized_extensions = "mp4|mp3|pdf|jpg"
+    message = Fabricate(:chat_message, user: user1, chat_channel: channel, message: "")
+    video = Fabricate(:upload, original_filename: "test_video.mp4", extension: "mp4")
+    audio = Fabricate(:upload, original_filename: "test_audio.mp3", extension: "mp3")
+    attachment = Fabricate(:upload, original_filename: "test_file.pdf", extension: "pdf")
+    image = Fabricate(:upload, width: 100, height: 200, original_filename: "test_img.jpg", extension: "jpg")
+    cu1 = ChatUpload.create(chat_message: message, created_at: 10.seconds.ago, upload: video)
+    cu2 = ChatUpload.create(chat_message: message, created_at: 9.seconds.ago, upload: audio)
+    cu3 = ChatUpload.create(chat_message: message, created_at: 8.seconds.ago, upload: attachment)
+    cu4 = ChatUpload.create(chat_message: message, created_at: 7.seconds.ago, upload: image)
+    video_markdown = UploadMarkdown.new(video).to_markdown
+    audio_markdown = UploadMarkdown.new(audio).to_markdown
+    attachment_markdown = UploadMarkdown.new(attachment).to_markdown
+    image_markdown = UploadMarkdown.new(image).to_markdown
+
+    expect(service(message.id).generate_markdown).to eq(<<~MARKDOWN)
+    [chat quote="martinchat;#{message.id};#{message.created_at.iso8601}" channel="The Beam Discussions"]
+    #{video_markdown}
+    #{audio_markdown}
+    #{attachment_markdown}
+    #{image_markdown}
+    [/chat]
+    MARKDOWN
```

Comment:
🧐 hmm should we be asserting the result of `UploadMarkdown`?

I guess the responsibility of this test is interesting. I'm thinking perhaps you can shorten this (to just one authorized extension and one non-authorized one, perhaps??) by creating unit tests for `chat_message#to_markdown`

Reply:
> asserting the result of UploadMarkdown

Nah I think that should be the responsibility of the UploadMarkdown test, which I added onto in core with https://github.com/discourse/discourse/pull/15930

Though I will add a unit test for `to_markdown`. The authorized extension stuff shouldn't matter too much either, the upload would not exist if the extension was unauthorized. I mainly wanted in this test to check that all the different types of uploads would get their markdown.